src/init.cpp

```diff
@@ -1812,5 +1815,24 @@ bool AppInitMain(InitInterfaces& interfaces)
         g_banman->DumpBanlist();
     }, DUMP_BANS_INTERVAL * 1000);
 
+    // Once initial sync is finished, flush state to protect against data loss
+    scheduler.scheduleEvery([]{
+        static bool did_initial_flush = false;
+        if (did_initial_flush || IsInitialBlockDownload()) {
+            return;
+        }
+
+        static int last_chain_height = -1;
+        LOCK(cs_main);
+        int current_height = chainActive.Height();
+        if (last_chain_height == -1 || last_chain_height != current_height) {
+            last_chain_height = current_height;
+            return;
+        }
+
+        did_initial_flush = true;
```

Comment:
Can't we just unschedule the task?

Reply:
Unless I'm missing something, I don't think it's possible to unschedule without making a lot more changes https://github.com/bitcoin/bitcoin/blob/master/src/scheduler.h#L58

Reply:
Use `scheduleFromNow` conditionally?

Reply:
Updated. How is that?